app/src/main/java/net/squanchy/schedule/SchedulePageView.java

```diff
@@ -59,12 +69,28 @@ protected void onFinishInflate() {
 
         progressBar = findViewById(R.id.progressbar);
 
-        ViewPager viewPager = findViewById(R.id.viewpager);
+        viewPager = findViewById(R.id.viewpager);
         TabLayout tabLayout = findViewById(R.id.tabstrip);
         tabLayout.setupWithViewPager(viewPager);
         hackToApplyTypefaces(tabLayout);
 
         viewPager.setAdapter(viewPagerAdapter);
+        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
+            @Override
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+                //nothing to do
+            }
+
+            @Override
+            public void onPageSelected(int position) {
+                currentPage = position;
```

Comment:
Isn't this triggered when calling `viewPager.setCurrentItem()`? I never remember lol

Reply:
It might! It was late night and I didn't bother to manual test nor write test.